tests/ImageSharp.Tests/Primitives/Matrix5x4Tests.cs

```diff
@@ -0,0 +1,270 @@
+            m.M24 = (value1.M21 * value2.M14) + (value1.M22 * value2.M24) + (value1.M23 * value2.M34) + (value1.M24 * value2.M44);
+
+            // Third row
+            m.M31 = (value1.M31 * value2.M11) + (value1.M32 * value2.M21) + (value1.M33 * value2.M31) + (value1.M34 * value2.M41);
+            m.M32 = (value1.M31 * value2.M12) + (value1.M32 * value2.M22) + (value1.M33 * value2.M32) + (value1.M34 * value2.M42);
+            m.M33 = (value1.M31 * value2.M13) + (value1.M32 * value2.M23) + (value1.M33 * value2.M33) + (value1.M34 * value2.M43);
+            m.M34 = (value1.M31 * value2.M14) + (value1.M32 * value2.M24) + (value1.M33 * value2.M34) + (value1.M34 * value2.M44);
+
+            // Fourth row
+            m.M41 = (value1.M41 * value2.M11) + (value1.M42 * value2.M21) + (value1.M43 * value2.M31) + (value1.M44 * value2.M41);
+            m.M42 = (value1.M41 * value2.M12) + (value1.M42 * value2.M22) + (value1.M43 * value2.M32) + (value1.M44 * value2.M42);
+            m.M43 = (value1.M41 * value2.M13) + (value1.M42 * value2.M23) + (value1.M43 * value2.M33) + (value1.M44 * value2.M43);
+            m.M44 = (value1.M41 * value2.M14) + (value1.M42 * value2.M24) + (value1.M43 * value2.M34) + (value1.M44 * value2.M44);
+
+            // Fifth row
+            m.M51 = (value1.M51 * value2.M11) + (value1.M52 * value2.M21) + (value1.M53 * value2.M31) + (value1.M54 * value2.M41) + value2.M51;
+            m.M52 = (value1.M51 * value2.M12) + (value1.M52 * value2.M22) + (value1.M53 * value2.M32) + (value1.M54 * value2.M52) + value2.M52;
+            m.M53 = (value1.M51 * value2.M13) + (value1.M52 * value2.M23) + (value1.M53 * value2.M33) + (value1.M54 * value2.M53) + value2.M53;
+            m.M54 = (value1.M51 * value2.M14) + (value1.M52 * value2.M24) + (value1.M53 * value2.M34) + (value1.M54 * value2.M54) + value2.M54;
+
+            Assert.Equal(m, value1 * value2, this.ApproximateFloatComparer);
+        }
+
+        [Fact]
+        public void Matrix5x4MultiplyScalar()
+        {
+            Matrix5x4 m = this.CreateAllTwos();
+            Assert.Equal(this.CreateAllFours(), m * 2, this.ApproximateFloatComparer);
+        }
+
+        [Fact]
+        public void Matrix5x4Subtract()
+        {
+            Matrix5x4 m = this.CreateAllOnes() + this.CreateAllTwos();
+            Assert.Equal(this.CreateAllThrees(), m);
+        }
+
+        [Fact]
+        public void Matrix5x4Negate()
+        {
+            Matrix5x4 m = this.CreateAllOnes() * -1F;
+            Assert.Equal(m, -this.CreateAllOnes());
+        }
+
+        [Fact]
+        public void Matrix5x4Add()
+        {
+            Matrix5x4 m = this.CreateAllOnes() + this.CreateAllTwos();
+            Assert.Equal(this.CreateAllThrees(), m);
+        }
+
+        [Fact]
+        public void Matrix5x4HashCode()
+        {
+#if NETCOREAPP2_1
+            Matrix5x4 m = KnownFilterMatrices.CreateBrightnessFilter(.5F);
+            HashCode hash = default;
```

Comment:
I would not add such a strict test case for `GetHashCode()`. The only thing we should ensure is that for equal objects hash code is also equal. (The opposite is not true!)

Reply:
I just copied what they do with Numerics tests. I think it still adds value as it ensures we touch every field when calculating the hashcode

Reply:
Ok, let's keep it as is for now!